clang/lib/StaticAnalyzer/Core/ExprEngine.cpp

```diff
@@ -2556,10 +2556,19 @@ void ExprEngine::processCFGBlockEntrance(const BlockEdge &L,
     const Stmt *Term = nodeBuilder.getContext().getBlock()->getTerminatorStmt();
     if (!isa_and_nonnull<ForStmt, WhileStmt, DoStmt, CXXForRangeStmt>(Term))
       return;
+
+    // FIXME:
```

Comment:
Please have a look at this workaround. I've noticed that, after adding back the assertion, the CFG element from the `ExprEngine` was incorrect. Therefore this workaround is put in place as a replacement for the terminator.

Reply:
Now thinking about this, and looking at the assertion within `getWidenedLoopState` we don't actually need to assert that the wrapped Stmt is a loop statement. As for correctness, it might be surprising to hear but it also doesn't really matter what CFGElement we would pass, the important part to pass something, and do so deterministically.

So if it makes the code simpler, we can drop the assertion inside, and even pass the wrong CFGElement here.
Would it make the patch simpler?

Reply:
Fair enough, I will remove the assertion but leave this FIXME here for future reference.

Reply:
I suspect we need to actually provide a valid CFG element here, so maybe we need to pass the first element in the new CFG Block?